src/main/java/com/cloudbees/jenkins/plugins/bitbucket/ForkPullRequestDiscoveryTrait.java

```diff
@@ -0,0 +1,358 @@
+         */
+        @Override
+        public boolean checkTrusted(@NonNull SCMSourceRequest request, @NonNull ChangeRequestSCMHead2 head) {
+            return false;
+        }
+
+        /**
+         * Our descriptor.
+         */
+        @Extension
+        public static class DescriptorImpl extends SCMHeadAuthorityDescriptor {
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getDisplayName() {
+                return Messages.ForkPullRequestDiscoveryTrait_nobodyDisplayName();
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public boolean isApplicableToOrigin(@NonNull Class<? extends SCMHeadOrigin> originClass) {
+                return SCMHeadOrigin.Fork.class.isAssignableFrom(originClass);
+            }
+        }
+    }
+
+    /**
+     * An {@link SCMHeadAuthority} that trusts forks belonging to the same account.
+     */
+    public static class TrustTeamForks
+            extends SCMHeadAuthority<BitbucketSCMSourceRequest, PullRequestSCMHead, SCMRevision> {
+
+        /**
+         * Constructor.
+         */
+        @DataBoundConstructor
+        public TrustTeamForks() {
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        protected boolean checkTrusted(@NonNull BitbucketSCMSourceRequest request, @NonNull PullRequestSCMHead head)
+                throws IOException, InterruptedException {
+            if (!head.getOrigin().equals(SCMHeadOrigin.DEFAULT)) {
+                return head.getRepoOwner().equalsIgnoreCase(request.getRepoOwner());
+            }
+            return false;
+        }
+
+        /**
+         * Our descriptor.
+         */
+        @Extension
+        public static class DescriptorImpl extends SCMHeadAuthorityDescriptor {
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getDisplayName() {
+                return Messages.ForkPullRequestDiscoveryTrait_teamDisplayName();
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public boolean isApplicableToOrigin(@NonNull Class<? extends SCMHeadOrigin> originClass) {
+                return SCMHeadOrigin.Fork.class.isAssignableFrom(originClass);
+            }
+
+        }
+    }
+
+    /**
+     * An {@link SCMHeadAuthority} that trusts everyone.
+     */
+    public static class TrustEveryone extends SCMHeadAuthority<SCMSourceRequest, ChangeRequestSCMHead2, SCMRevision> {
+        /**
+         * Constructor.
+         */
+        @DataBoundConstructor
+        public TrustEveryone() {
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        protected boolean checkTrusted(@NonNull SCMSourceRequest request, @NonNull ChangeRequestSCMHead2 head) {
+            return !head.getOrigin().equals(SCMHeadOrigin.DEFAULT);
```

Comment:
It's only called on `SCMHeadOrigin.Fork.class.isAssignableFrom(originClass)`  so I will refactor to `true`